units/recalculate_change_points.go

```diff
@@ -113,24 +113,58 @@ func (j *recalculateChangePointsJob) Run(ctx context.Context) {
 		}
 
 		result, err := j.changePointDetector.DetectChanges(ctx, floatSeries)
+		if err != nil {
+			j.AddError(errors.Wrapf(err, "Unable to detect change points in time perfData %s", j.PerformanceResultId))
+			return
+		}
 
 		var changePoints []model.ChangePoint
-		for _, pointIndex := range result {
-			mapped := model.CreateChangePoint(pointIndex+latestTriagedChangePointIndex, perfData.Measurement, j.changePointDetector.Algorithm().Name(), j.changePointDetector.Algorithm().Version(), algorithmConfigurationToOptions(j.changePointDetector.Algorithm().Configuration()))
+		for i, pointIndex := range result {
```

Comment:
can/should this math be a method on the change point?

Reply:
No I don't think so. This for loop is calculating multiple ChangePoints so adding this whole calculation as a method on a single ChangePoint struct doesn't seem to be a good match. 

Adding just the percent_change calculation as a method doesn't seem to be a good match either as the calculation requires a lot of state to do the calculation for each region. So the actual signature for the method would be pretty unwieldy and wouldn't add much benefit imo.